File: CarRental.DAL/Models/Registration.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace CarRental.Database.Models
{
    public enum RegistrationType
    {
        PickUp,
        Return
    }

    public class Registration
    {
        public int Id { get; set; }
        public RegistrationType RegistrationType { get; set; }
        [DataType(DataType.DateTime)]
        [DisplayFormat(DataFormatString = "{0:yyyy-MM-dd HH:mm}", ApplyFormatInEditMode = true)]
        public DateTime DateTime { get; set; }
        public int DistanceMeter { get; set; }
    }
}

File: CarRental.Web/ClientApp/src/app/models/registration.ts
export class Registration {
  id: number;
  registrationType: RegistrationType;
  dateTime: string;
  distanceMeter: number;

  constructor(registrationType: RegistrationType, date: string, distanceMeter: number) {
    this.registrationType = registrationType;
    this.dateTime = date;
    this.distanceMeter = distanceMeter;
  }
}

export enum RegistrationType {
  PickUp = 0,
  Return = 1
}
